homeassistant/components/elkm1/__init__.py

```diff
@@ -0,0 +1,206 @@
+"""
+Support the ElkM1 Gold and ElkM1 EZ8 alarm / integration panels.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/elkm1/
+"""
+
+import logging
+
+import voluptuous as vol
+from homeassistant.const import (
+    CONF_EXCLUDE, CONF_HOST, CONF_INCLUDE, CONF_PASSWORD,
+    CONF_TEMPERATURE_UNIT, CONF_USERNAME, TEMP_FAHRENHEIT)
+from homeassistant.core import HomeAssistant, callback  # noqa
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType  # noqa
+
+DOMAIN = "elkm1"
+
+REQUIREMENTS = ['elkm1-lib==0.7.10']
+
+CONF_AREA = 'area'
+CONF_COUNTER = 'counter'
+CONF_KEYPAD = 'keypad'
+CONF_OUTPUT = 'output'
+CONF_SETTING = 'setting'
+CONF_TASK = 'task'
+CONF_THERMOSTAT = 'thermostat'
+CONF_PLC = 'plc'
+CONF_ZONE = 'zone'
+CONF_ENABLED = 'enabled'
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_DOMAINS = ['alarm_control_panel']
+
+
+def _host_validator(config):
+    """Validate that a host is properly configured."""
+    if config[CONF_HOST].startswith('elks://'):
+        if CONF_USERNAME not in config or CONF_PASSWORD not in config:
+            raise vol.Invalid("Specify username and password for elks://")
+    elif not config[CONF_HOST].startswith('elk://') and not config[
+            CONF_HOST].startswith('serial://'):
+        raise vol.Invalid("Invalid host URL")
+    return config
+
+
+def _elk_range_validator(rng):
+    def _elk_value(val):
+        from elkm1_lib.message import housecode_to_index
+        return int(val) if val.isdigit() else (housecode_to_index(val) + 1)
+
+    vals = [s.strip() for s in str(rng).split('-')]
+    start = _elk_value(vals[0])
+    end = start if len(vals) == 1 else _elk_value(vals[1])
+    return (start, end)
+
+
+CONFIG_SCHEMA_SUBDOMAIN = vol.Schema({
+    vol.Optional(CONF_ENABLED, default=True): cv.boolean,
+    vol.Optional(CONF_INCLUDE, default=[]): [_elk_range_validator],
```

Comment:
We can't use the validator here since it uses the requirements library which might not be installed yet.